### ISSUE TITLE
Validation errors when creating customers

### ISSUE THREAD

**Issue by lancedockins** [helpscout/helpscout-api-php#164]
For some reason creating customers with an Email or Phone object will throw a validation error.  We even tried doing this with the test code from the examples in this PHP API.  

Here's a sample:

	`require dirname(__FILE__) . '/config/helpscout.php';
	$client = ApiClientFactory::createClient();
	$client->useClientCredentials($appId, $appSecret);			
	
	$filters = (new CustomerFilters())->withQuery('email:"' . $data['email'] . '"');
	$customers = $client->customers()->list($filters);

	if(count($customers) == 0) {
		// Create Customer
		$customer = new Customer();
		$customer->setFirstName($data['firstName']);
		$customer->setLastName($data['lastName']);

		if(!empty($data['email'])) {
			$customer->addEmail(new Email([
			    'email' => "my-customer@their-business.com",
			    'type' => 'work',
			]));
		}

		$test = $client->customers()->create($customer);
		var_dump($test);
	}`

As long as we don't add the Email object via addEmail, the call will succeed.  If we use this same method to add a phone, it will also throw a validation error.  We have tested this with other email addresses, phone numbers, etc with multiple formats and get the same validation error every time.  Seems to be returning a 400 error from the API under the hood.

The $data object is populated with a sample name and email address.

This is the error that is produced when no customer is returned and the code tries to create a new customer. As long as we do not try to add and email or phone, the above code works. 
As soon as we try to add an email or phone, it fails.  

Note that the path has been sanitized in the error message for security reasons.  The actual path is not /path/to/code obviously.

`<b>Fatal error</b>: Uncaught HelpScout\Api\Exception\ValidationErrorException: Validation error in /path/to/code/includes/helpscout/vendor/helpscout/api/src/Http/Handlers/ValidationHandler.php:23
Stack trace:
#0 /path/to/code/includes/helpscout/vendor/guzzlehttp/promises/src/Promise.php(203): HelpScout\Api\Http\Handlers\ValidationHandler-&gt;HelpScout\Api\Http\Handlers\{closure}(Object(GuzzleHttp\Psr7\Response))
#1 /path/to/code/includes/helpscout/vendor/guzzlehttp/promises/src/Promise.php(156): GuzzleHttp\Promise\Promise::callHandler(1, Object(GuzzleHttp\Psr7\Response), Array)
#2 /path/to/code/includes/helpscout/vendor/guzzlehttp/promises/src/TaskQueue.php(47): GuzzleHttp\Promise\Promise::GuzzleHttp\Promise\{closure}()
#3 /path/to/code/includes/helpscout/vendor/guzzlehttp/promises/src/Promise.php(246): GuzzleHttp\Promise\TaskQueue-&gt;run(true)
#4 /home/wpboost/www/wp-content/themes/wpb in <b>/path/to/code/includes/helpscout/vendor/helpscout/api/src/Http/Handlers/ValidationHandler.php</b> on line <b>23</b><br />
`

**Comment by bkuhl** [comment]
What you'll want to do is [catch the exception similar to this example](https://github.com/helpscout/helpscout-api-php#validation) and look at the errors to see why the API is rejecting the request.

**Comment by lancedockins** [comment]
This is what comes back.   It's complaining about the email value being empty.  Considering that the example that I used is quite literally from the documentation, it would seem like the PHP object is not hydrating properly given the input.  Either that or the documentation/examples are wrong.

Any suggestions on how to address this?

`string(45) "ec159308-58f3-42b7-827a-2788227a7d8c#41821721"
array(2) {
  [0]=>
  object(HelpScout\Api\Http\Hal\VndError)#2103 (4) {
    ["message":"HelpScout\Api\Http\Hal\VndError":private]=>
    string(17) "must not be empty"
    ["logRef":"HelpScout\Api\Http\Hal\VndError":private]=>
    NULL
    ["path":"HelpScout\Api\Http\Hal\VndError":private]=>
    string(14) "emails[0].type"
    ["errors":"HelpScout\Api\Http\Hal\VndError":private]=>
    array(0) {
    }
  }
  [1]=>
  object(HelpScout\Api\Http\Hal\VndError)#2104 (4) {
    ["message":"HelpScout\Api\Http\Hal\VndError":private]=>
    string(17) "must not be empty"
    ["logRef":"HelpScout\Api\Http\Hal\VndError":private]=>
    NULL
    ["path":"HelpScout\Api\Http\Hal\VndError":private]=>
    string(15) "emails[0].value"
    ["errors":"HelpScout\Api\Http\Hal\VndError":private]=>
    array(0) {
    }
  }
}
`

**Comment by bkuhl** [comment]
It seems the example is incorrect, it should be:

```
$email = new Email();
$email->hydrate([
    'value' => "my-customer@their-business.com",
    'type' => 'work',
]);
```

I've submitted a [pull request to allow `addEmail()`](https://github.com/helpscout/helpscout-api-php/pull/165) to accept 2 strings, rather than having to build an `Email`.

**Comment by lancedockins** [comment]
First things first, thank you.

That said, did you happen to duplicate this shorthand change for Phone objects or other similar objects?  In my testing so far, this problem was not limited just to Email objects.  I ran into the same behavior with Phone objects as well.  

Given that it extends to both Phone and Email, I would have to assume that it would extend to Chat, Website, etc

**Comment by bkuhl** [comment]
Yep, you're right, those should have these shorthand methods as well.  I've added them in PR https://github.com/helpscout/helpscout-api-php/pull/171

**Comment by bkuhl** [comment]
Now available in [v2.3.5](https://github.com/helpscout/helpscout-api-php/releases/tag/2.3.5)

**Comment by lancedockins** [comment]
Thank you, Ben!
